### ISSUE TITLE
Move to JLine 2.7

### ISSUE THREAD

**Issue by inkytonik** [sbt/sbt#523]
In the Kiama library we've moved to JLine 2.7, which appears to be the most recent and maintained version of JLine around.

All is reasonably well, except that we've just discovered an incompatibility with sbt 0.12.0 which uses JLine 1.0. When we run a program that uses JLine it works fine, but when we come back to sbt echo is turned off at the sbt command prompt.

The problem occurs if the program we run uses JLine 2.7, but not if it uses JLine 1. The projects we are testing use 

```
fork in run := true

connectInput in run := true

outputStrategy in run := Some (StdoutOutput)
```

to avoid a clash between the two JLine library versions in a single JVM.

I'm running on a Mac OS X 10.8 with Java 1.7.0_06, but I have a colleague who has seen the same problem on a Windows box.

The cut-down sbt project at

```
https://bitbucket.org/inkytonik/scratch-jline-problem
```

illustrates the problem. src/Main.scala contains two implementations of JLine-based reading of lines (based on the sbt code for the same purpose). When the JLine 2.7 version is used and the build.sbt calls for JLine 2.7, we see the echo off after the program runs (i.e., no characters are echoed when we type). When the JLine 1 version is used and the build.sbt calls for JLine 1.0, the sbt command line behaves as we would expect.

It is possible that we are using JLine 2.7 incorrectly, but the code seems ok. Perhaps there is a bug in JLine 2.7, but the terminal management works fine if we run the above project directly from the shell without using sbt.

So, our conclusion is that there is an incompatibility between the two JLine versions. Possibly a fix in sbt can remedy the problem, but the easiest approach would seem to be for sbt to move to the latest version.

**Comment by harrah** [comment]
I'm sure JLine 2.x and 1.0 are incompatible and we can see what needs to be done to fix this problem, but it isn't likely that sbt will move to 2.7 any time soon.

**Comment by inkytonik** [comment]
Thanks for taking a look at it. I appreciate that moving sbt to the new version is a non-trivial change. But one that I expect you will make eventually, since 1.0 is standing still whereas 2.x is being actively developed.

BTW, if you have any clues to offer towards a fix, I'm happy to include code in our JLine interface to make this work; just don't know what else to try.

**Comment by inkytonik** [comment]
FYI, for the next Kiama release we will move back to JLine 1.0. The incompatibility with sbt is more of an issue for us than being with a slightly older than cutting edge JLine.

**Comment by harrah** [comment]
Implemented by #684, which uses JLine 2.10.
